### PR TITLE
Implement state peek, backup function and defaultStateFn

### LINKED ISSUE #53: [lex] implement defaultStateFn
**Subject**
lex

**Detail**
Implement defaultStateFn function and test cases.

### PR DESCRIPTION
**Pull request by junbeomlee**
resolved: #53 #54

details:

[task detail...]

1. Implement state peek function
2. Implement state backup function
3. Implement defaultStateFn function

- [x] Test case

### CODE REVIEW

**Review by boohyunsik — APPROVED** [review]
Good!

File: parse/lex.go [hunk 1]
```diff
@@ -157,16 +164,92 @@ func (s *state) acceptRun(valid string) {
...
+//	Rbrace // }
+//
+// TODO: LTE, GTE, spaceStateFn, numberStateFn, identifierStateFn case
+func defaultStateFn(s *state, e emitter) stateFn {
+
+	switch ch := s.next(); {
+
+	case ch == '!':
+		if s.peek() == '=' {
+			s.next()
```
Comment (zeroFruit) [inline parse/lex.go:198]: How about refactoring this logic to another func?

Comment (junbeomlee) [reply parse/lex.go:198]: Because this is so simple logic,
i think this seems to be more understandable!

File: parse/lex_internal_test.go [hunk 2]
```diff
@@ -102,3 +102,144 @@ func TestState_next(t *testing.T) {
 			s.line)
 	}
 }
+
+func TestState_backup(t *testing.T) {
+	input := "hello\n"
```
Comment (zeroFruit) [inline parse/lex_internal_test.go:107]: just make sure this really works, plz add multi line test  👍

**Review by zeroFruit — APPROVED** [review]

**Pull request merged**